internal/services/capacity/datasource_tenant_capacity.go

```diff
@@ -54,11 +54,11 @@ type ConsumptionDataSourceModel struct {
 	RatedUpdatedOn  types.String  `tfsdk:"rated_updated_on"`
 }
 
-func (d *TenantCapacityDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+func (d *DataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
```

Comment:
We should fix this by calling our `enterRequestContext` method which needs context.  Making context a _ seems like an antipattern we'd want to avoid